server/repository/src/db_diesel/assets/asset_catalogue_item.rs

```diff
@@ -173,6 +139,29 @@ fn create_filtered_query(filter: Option<AssetCatalogueItemFilter>) -> BoxedAsset
         );
         apply_equal_filter!(query, class_id, asset_catalogue_item_dsl::asset_class_id);
         apply_equal_filter!(query, type_id, asset_catalogue_item_dsl::asset_type_id);
+
+        if let Some(class_filter) = class {
+            let mut sub_query = asset_class_dsl::asset_class
+                .select(asset_class_dsl::id)
+                .into_boxed();
+            apply_string_filter!(sub_query, Some(class_filter), asset_class_dsl::name);
+            query = query.filter(asset_catalogue_item_dsl::asset_class_id.eq_any(sub_query));
+        }
+
+        if let Some(r#type_filter) = r#type {
+            let mut sub_query = asset_type_dsl::asset_type
+                .select(asset_type_dsl::id)
+                .into_boxed();
+            apply_string_filter!(sub_query, Some(r#type_filter), asset_type_dsl::name);
+            query = query.filter(asset_catalogue_item_dsl::asset_type_id.eq_any(sub_query));
+        }
+        if let Some(category_filter) = category {
+            let mut sub_query = asset_category_dsl::asset_category
+                .select(asset_category_dsl::id)
+                .into_boxed();
+            apply_string_filter!(sub_query, Some(category_filter), asset_category_dsl::name);
+            query = query.filter(asset_catalogue_item_dsl::asset_category_id.eq_any(sub_query));
+        }
```

Comment:
@mark-prins if you were curious, here is where we moved the new filters (to within create_filtered_query)